rustls/src/common_state.rs

```diff
@@ -140,6 +142,17 @@ impl CommonState {
         self.negotiated_version
     }
 
+    /// Which kind of handshake was performed.
+    ///
+    /// This tells you whether the handshake was a resumption or not.
+    ///
+    /// This will return `Err(Error::HandshakeNotComplete)` before it is
+    /// known which sort of handshake occurred.
+    pub fn handshake_kind(&self) -> Result<HandshakeKind, Error> {
```

Comment:
I'm a little surprised at the choice of `Result` here? IIRC most of the similar methods just return `Option`, which seems appropriate here, too?